okhttp/src/main/java/okhttp3/internal/Util.kt

```diff
@@ -113,21 +112,20 @@ fun intersect(
 }
 
 /**
- * Returns true if there is an element in [first] that is also in [second]. This
+ * Returns true if there is an element in [this] that is also in [other]. This
  * method terminates if any intersection is found. The sizes of both arguments are assumed to be
  * so small, and the likelihood of an intersection so great, that it is not worth the CPU cost of
  * sorting or the memory cost of hashing.
  */
-fun nonEmptyIntersection(
-  comparator: Comparator<String>,
-  first: Array<String>?,
-  second: Array<String>?
+fun Array<String>.hasIntersection(
+  other: Array<String>?,
+  comparator: Comparator<in String>
```

Comment:
naturalOrder is a compelling default for these.

Reply:
Yeah I was debating whether to do that or not. They're actually only used in one place where a custom comparator is always specified so I elected not to.